tests/src/Kernel/Access/OgGroupContentOperationAccessTest.php

```diff
@@ -0,0 +1,357 @@
+    $this->installEntitySchema('node');
+    $this->installEntitySchema('og_membership');
+    $this->installEntitySchema('user');
+    $this->installSchema('system', 'sequences');
+
+    $this->entityTypeManager = $this->container->get('entity_type.manager');
+
+    $this->groupBundle = Unicode::strtolower($this->randomMachineName());
+
+    // Create a test user with UID 1. This user has universal access.
+    $this->users['uid1'] = User::create(['name' => $this->randomString()]);
+    $this->users['uid1']->save();
+
+    // Create a user that will serve as the group owner. There are no special
+    // permissions granted to the group owner, so this user can be used for
+    // creating entities that are not owned by the user under test.
+    $this->users['group_owner'] = User::create(['name' => $this->randomString()]);
+    $this->users['group_owner']->save();
+
+    // Declare that the test entity is a group type.
+    Og::groupManager()->addGroup('entity_test', $this->groupBundle);
+
+    // Create the test group.
+    $this->group = EntityTest::create([
+      'type' => $this->groupBundle,
+      'name' => $this->randomString(),
+      'user_id' => $this->users['group_owner']->id(),
+    ]);
+    $this->group->save();
+
+    // Create 3 test roles with associated permissions. We will simulate a
+    // project that has two group content types:
+    // - 'newsletter_subscription': Any registered user can create entities of
+    //   this type, even if they are not a member of the group.
+    // - 'article': These can only be created by group members. Normal members
+    //   can edit and delete their own articles, while admins can edit and
+    //   delete any article.
+    $permission_matrix = [
+      OgRoleInterface::ANONYMOUS => [
+        'create newsletter_subscription comment',
+        'update own newsletter_subscription comment',
+        'delete own newsletter_subscription comment',
+      ],
+      OgRoleInterface::AUTHENTICATED => [
+        'create newsletter_subscription comment',
+        'update own newsletter_subscription comment',
+        'delete own newsletter_subscription comment',
+        'create article content',
+        'edit own article content',
+        'delete own article content',
+      ],
+      // The administrator is not explicitly granted permission to edit or
+      // delete their own group content. Having the 'any' permission should be
+      // sufficient to also be able to edit their own content.
+      OgRoleInterface::ADMINISTRATOR => [
+        'create newsletter_subscription comment',
+        'update any newsletter_subscription comment',
+        'delete any newsletter_subscription comment',
+        'create article content',
+        'edit any article content',
+        'delete any article content',
+      ],
+    ];
+
+    foreach ($permission_matrix as $role_name => $permissions) {
+      $role_id = "{$this->group->getEntityTypeId()}-{$this->group->bundle()}-$role_name";
```

Comment:
I created this method. I called it `OgRole::loadByGroupAndName()` to be consistent with the other methods that allow to use the role name, such as `OgRole::setName()`.

We had a similar helper function before: `Og::getRole()` but this took the entity type ID, bundle and role name as parameters. I removed it because this one makes a lot more sense: it is in `OgRole` where it belongs, and allowing to pass in a group entity is much better DX than having to extract the entity ID and bundle on the calling side.

I grepped the entire codebase to find where we were using this pattern and replaced it with the new method.